code/modules/power/apc.dm

```diff
@@ -146,30 +152,24 @@ GLOBAL_LIST_EMPTY(apcs)
 	if(terminal)
 		terminal.connect_to_network()
 
-/obj/machinery/power/apc/drain_power(var/drain_check, var/surge, var/amount = 0)
+/obj/machinery/power/apc/drain_energy(datum/acter, amount, flags)
+	charging = FALSE
+	// makes sure fully draining apc cell won't break cell charging
 
-	if(drain_check)
-		return 1
-
-	//This makes sure fully draining an APC cell won't break the cell charging.
-	charging = 0
-
-	var/drained_energy = 0
+	var/drained = 0
 
-	//Draws power from the grid first, if available.
-	//This is like draining from a cable, so nins and
-	//twizs can do that without having to pry floortiles.
-	if(terminal && terminal.powernet)
+	if(terminal?.powernet)
 		terminal.powernet.trigger_warning()
-		drained_energy += terminal.powernet.draw_power(amount)
+		// no conversion - amount = kj, draw_power is in kw
+		drained += terminal.powernet.draw_power(amount)
 
 	//The grid rarely gives the full amount requested, or perhaps the grid
 	//isn't connected (wire cut), in either case we draw what we didn't get
 	//from the cell instead.
```

Comment:
Maybe double-check that your subtraction of power from the grid actually did anything, if possible. If it didn't, draw only from the cell. (if the cell is below, i'unno, half full, pretend it did and refill it anyways. At that point precision is too far gone to be saveable.)

This'll allow you to avoid APCs using 0 power when precision gets too out of wack. However considering you redid the scaling it's likely unnecessary, in which case disregard.